controllers/controllers/services/instances/managed/controller.go

```diff
@@ -129,6 +121,14 @@ func (r *Reconciler) ReconcileResource(ctx context.Context, serviceInstance *kor
 		return ctrl.Result{}, fmt.Errorf("failed to create client for broker %q: %w", serviceBroker.Name, err)
 	}
 
+	if isReady(serviceInstance) {
```

Comment:
Nitpick: I would move the ready/failed checks before getting the plan and creating the osbapi client - in case we have reached the terminal state, there is nothing to do, therefore no need to create the osbapi client